go/logic/migrator.go

```diff
@@ -1279,30 +1279,31 @@ func (this *Migrator) executeWriteFuncs() error {
 					return err
 				}
 			}
-		default:
-			{
+		case copyRowsFunc := <-this.copyRowsQueue:
+		priority:
+			for {
 				select {
-				case copyRowsFunc := <-this.copyRowsQueue:
+				case eventStruct := <-this.applyEventsQueue:
 					{
-						copyRowsStartTime := time.Now()
-						// Retries are handled within the copyRowsFunc
-						if err := copyRowsFunc(); err != nil {
-							return this.migrationContext.Log.Errore(err)
-						}
-						if niceRatio := this.migrationContext.GetNiceRatio(); niceRatio > 0 {
-							copyRowsDuration := time.Since(copyRowsStartTime)
-							sleepTimeNanosecondFloat64 := niceRatio * float64(copyRowsDuration.Nanoseconds())
-							sleepTime := time.Duration(int64(sleepTimeNanosecondFloat64)) * time.Nanosecond
-							time.Sleep(sleepTime)
+						if err := this.onApplyEventStruct(eventStruct); err != nil {
```

Comment:
```suggestion
						this.throttler.throttle(nil)
						if err := this.onApplyEventStruct(eventStruct); err != nil {
```
I believe we should check the throttler before each `this.onApplyEventStruct(eventStruct)`